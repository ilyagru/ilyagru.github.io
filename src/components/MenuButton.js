import React from 'react';
import PT from 'prop-types';
import styled, { css } from 'styled-components';

import { enableScroll, disableScroll } from '../utils/window';

const BaseButton = styled.button`
  border: 1px solid ${({ theme }) => theme.accentColor};
  background-color: transparent;
  outline: 0;
  font-family: ${({ theme }) => theme.merriweatherFont};
  transition: ${({ theme }) => theme.transition};
  position: relative;
  z-index: 20;
  color: ${({ theme }) => theme.textColor};
  font-weight: 800;

  ${({ isMenuOpen }) =>
    isMenuOpen
      ? css`
          transform: translate(15px) rotate(-90deg);
          background-color: ${({ theme }) => theme.accentColor};
          color: ${({ theme }) => theme.textColorReversed};
        `
      : css`
          transform: rotate(0deg);
        `}
`;

function MenuButton({ isMenuOpen, onClick }) {
  isMenuOpen ? disableScroll() : enableScroll();

  return (
    <BaseButton isMenuOpen={isMenuOpen} onClick={onClick}>
      menu
    </BaseButton>
  );
}

MenuButton.defaultProps = {
  isMenuOpen: false,
};

MenuButton.propTypes = {
  isMenuOpen: PT.bool,
  onClick: PT.func.isRequired,
};

export default MenuButton;
