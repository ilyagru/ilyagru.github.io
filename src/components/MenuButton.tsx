import React from 'react';
import styled, { css } from 'styled-components';
import { enableScroll, disableScroll } from '../utils/window';

type BaseButtonProps = {
  isMenuOpen: boolean;
};

const BaseButton = styled.button<BaseButtonProps>`
  border: 1px solid ${({ theme }) => theme.accentColor};
  background-color: transparent;
  outline: 0;
  font-family: ${({ theme }) => theme.merriweatherFont};
  transition: ${({ theme }) => theme.transition};
  position: relative;
  z-index: 20;
  color: ${({ theme }) => theme.textColor};
  font-weight: 800;

  ${({ isMenuOpen }) =>
    isMenuOpen
      ? css`
          transform: translate(15px) rotate(-90deg);
          background-color: ${({ theme }) => theme.accentColor};
          color: ${({ theme }) => theme.textColorReversed};
        `
      : css`
          transform: rotate(0deg);
        `}
`;

type Props = {
  isMenuOpen?: boolean;
  onClick: () => void;
};

function MenuButton({ isMenuOpen = false, onClick }: Props) {
  if (isMenuOpen) {
    disableScroll();
  } else {
    enableScroll();
  }

  return (
    <BaseButton isMenuOpen={isMenuOpen} onClick={onClick}>
      menu
    </BaseButton>
  );
}

export default MenuButton;
