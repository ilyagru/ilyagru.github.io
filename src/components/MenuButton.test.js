import React from 'react';
import renderer from 'react-test-renderer';

import MenuButton from './MenuButton';

describe('MenuButton', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MenuButton onClick={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
