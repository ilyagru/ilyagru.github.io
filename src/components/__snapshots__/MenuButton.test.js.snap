// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton renders correctly 1`] = `
<button
  className="sc-bdvvtL irLYvN"
  onClick={[MockFunction]}
>
  menu
</button>
`;
